{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "%matplotlib inline\n",
    "\n",
    "import seaborn as sns\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "import matplotlib\n",
    "\n",
    "from Bio import SeqIO\n",
    "from matplotlib import pyplot as plt\n",
    "from os import path\n",
    "from scipy.stats import binom\n",
    "from utils import qual_palette"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# Genome and pool data filenames\n",
    "genome_fname = \"../data/berkeleylab-feba-7de9bcaf0960/bin/g/Halo/genome.fna\"\n",
    "pool_fname1 = \"../data/HnTnSeqAnalysis/DSJD02HTn1_R1/DSJDo2Htn1_S1_l006_R1_001.pool\"\n",
    "pool_fname2 = \"../data/HnTnSeqAnalysis/DSJD03Htn2_R1/DSJD03Htn2_S2_l006_R1_001.pool\"\n",
    "\n",
    "gene_stats_fname1 = '../data/HnTnSeqAnalysis/geneStatsRep1.csv'\n",
    "gene_stats_fname2 = '../data/HnTnSeqAnalysis/geneStatsRep2.csv'\n",
    "\n",
    "# DataFrames for pool and gene data\n",
    "pool_df1 = pd.read_csv(pool_fname1, sep='\\t')\n",
    "pool_df2 = pd.read_csv(pool_fname2, sep='\\t')\n",
    "rep_df1 = pd.read_csv(gene_stats_fname1)\n",
    "rep_df2 = pd.read_csv(gene_stats_fname2)\n",
    "\n",
    "# DataFrame for essentiality calls \n",
    "essentiality_fname = '../data/essentiality_calls.csv'\n",
    "essentiality_df = pd.read_csv(essentiality_fname)\n",
    "\n",
    "# Grab genome length\n",
    "genome_size_bp = None\n",
    "for req in SeqIO.parse(genome_fname, \"fasta\"):\n",
    "    genome_size_bp = len(req)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "# When we are plotting the library we want to include insertions mapped in either replicate.\n",
    "# Therefore we merge the two dataframes. \n",
    "merge_on = pool_df1.columns.tolist()\n",
    "total_pool_df = pool_df1.merge(pool_df2, on=merge_on, how='outer')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAjwAAABGCAYAAADIKU4UAAAABHNCSVQICAgIfAhkiAAAAAlwSFlz\nAAALEgAACxIB0t1+/AAAAuFJREFUeJzt27+rT3Ecx/HPQcktGdxJud1BGdwb3ULhymiSTTJYTH6V\nulFsjCYlgywmWc0md7pXRK7hlqKrTBbJtdDxD7j3u/j0cV/n8RjPWV7v7Vnn++36vi8AAMk2tR4A\nAFCb4AEA4gkeACCe4AEA4gkeACCe4AEA4m1Z72XXdf6zDgBsGH3fd397vm7wlFLK01cf//2a/8TF\nq9fL+LEzrWdUc+DHw3L2ysnWM6q5PHe3TJza1XpGFSvPvpTl03daz6jmxtz1cmXneOsZ1Tw+Ml5u\nnT/aekY1d27fKzdPbG09o5pry+fKhbMXW8+o4tGTB2Xb1+nWM6q5//zSmu980gIA4gkeACDeoINn\nbGJf6wlVTR3a03pCVTv2bm89oZrk20op5dC2sdYTqprdv7v1hKpmJze3nlDVzNTB1hOqSb5tlIEH\nz1TrCVVNHxY8G1XybaWUcngsO3iOH5hoPaGq2cmRP//c0Gamc6Mg+bZRBh08AMAwCB4AIJ7gAQDi\nCR4AIJ7gAQDiCR4AIJ7gAQDiCR4AIJ7gAQDiCR4AIJ7gAQDiCR4AIJ7gAQDiCR4AIJ7gAQDiCR4A\nIJ7gAQDiCR4AIJ7gAQDiCR4AIJ7gAQDiCR4AIJ7gAQDiCR4AIJ7gAQDiCR4AIJ7gAQDiCR4AIJ7g\nAQDiCR4AIJ7gAQDiCR4AIJ7gAQDiCR4AIJ7gAQDiCR4AIJ7gAQDiCR4AIJ7gAQDiCR4AIJ7gAQDi\nCR4AIJ7gAQDiCR4AIJ7gAQDiCR4AIN6gg2d1Zan1hKreLXxoPaGqb8vfW0+oJvm2UkpZWF1tPaGq\nF29WWk+oav7Tr9YTqnr97mXrCdUk3zbKwIPnfesJVS0tCp6NKvm2UkpZ/JkdPPNvP7eeUNX8p9+t\nJ1T1eik3CpJvG2XQwQMADIPgAQDidX3fr/2y69Z+CQDwn+n7vvvb83WDBwAggU9aAEA8wQMAxBM8\nAEA8wQMAxBM8AEC8P+moYoWH1kkoAAAAAElFTkSuQmCC\n",
      "text/plain": [
       "<matplotlib.figure.Figure at 0x103aeb6d8>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "sns.palplot(qual_palette)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def makeFig2S1():\n",
    "    import pandas as pd\n",
    "    pathToData = \"./../RB-TnSeq/2017_11_03_miseq_barSeq1/\"\n",
    "    logratios_df1 = pd.read_csv(path.join(pathToData, 'fit_logratios.tab'), sep='\\t')\n",
    "    tValues_df1 = pd.read_csv(path.join(pathToData, 'fit_t.tab'), sep='\\t')\n",
    "    pathToData = \"./../RB-TnSeq/2017_11_03_miseq_barseq2/\"\n",
    "    logratios_df2 = pd.read_csv(path.join(pathToData, 'fit_logratios.tab'), sep='\\t')\n",
    "    tValues_df2 = pd.read_csv(path.join(pathToData, 'fit_t.tab'), sep='\\t')\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true,
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "# Making Panel D\n",
    "%matplotlib inline\n",
    "def makePanelD(ax):\n",
    "    import adjustText\n",
    "\n",
    "    import seaborn as sns\n",
    "    cols = ['#56AC52','#337848','#F2E92A','#F58320','#822F31','#CC3C39','#475BA6','#78CDDA','#5375BA','#EA3C81'\n",
    "            ,'#7E441D','#848585','#8E5EA7','#B9D36C','#000000']\n",
    "    colors = sns.color_palette(cols)    \n",
    "#     iCol=colors[3] #in operon\n",
    "    iCol=colors[9] #in operon\n",
    "    eCol=colors[4]  #known CCM genes\n",
    "    nCol=colors[7] #known non-CCM genes\n",
    "    aCol=colors[6]  #out of operon\n",
    "    \n",
    "    import pandas as pd\n",
    "    import numpy as np\n",
    "    from os import path\n",
    "\n",
    "    \n",
    "    pathToData = \"./../RB-TnSeq/2017_11_03_miseq_barSeq1/\"\n",
    "    gene_counts_df = pd.read_csv(path.join(pathToData, 'all.poolcount'), sep='\\t')\n",
    "    logratios_df = pd.read_csv(path.join(pathToData, 'fit_logratios.tab'), sep='\\t')\n",
    "    tValues_df = pd.read_csv(path.join(pathToData, 'fit_t.tab'), sep='\\t')\n",
    "    smaller_df = pd.DataFrame()\n",
    "    smaller_df[\"barcode\"] = gene_counts_df[\"barcode\"]\n",
    "    smaller_df[\"locusId\"] = gene_counts_df[\"locusId\"]\n",
    "    smaller_df[\"Initial\"] = gene_counts_df[\"test.S1\"]\n",
    "    smaller_df[\"HighCO2\"] = gene_counts_df[\"test.S2\"]\n",
    "    smaller_df[\"LowCO2\"] = gene_counts_df[\"test.S3\"]\n",
    "\n",
    "    # smaller_df\n",
    "    small_log_df = pd.DataFrame()\n",
    "    small_t_df = pd.DataFrame()\n",
    "    small_log_df[\"locusId\"] = logratios_df[\"locusId\"]\n",
    "    small_log_df[\"sysName\"] = logratios_df[\"sysName\"]\n",
    "    small_log_df[\"desc\"] = logratios_df[\"desc\"]\n",
    "    small_log_df[\"HighCO2Log\"] = logratios_df[\"setAS2 HCO\"]\n",
    "    small_log_df[\"LowCO2Log\"] = logratios_df[\"setAS3 LCO\"]\n",
    "\n",
    "    small_t_df[\"locusId\"] = tValues_df[\"locusId\"]\n",
    "    small_t_df[\"HighCO2T\"] = tValues_df[\"setAS2 HCO\"]\n",
    "    small_t_df[\"LowCO2T\"] = tValues_df[\"setAS3 LCO\"]\n",
    "    \n",
    "    geneLocs_df = pd.read_csv(path.join(pathToData, 'genes'), sep='\\t')\n",
    "#     cbbLocus_df = geneLocs_df[(geneLocs_df.begin>(983926-10000))&(geneLocs_df.begin<(983926+10000))]\n",
    "    cbbLocus_df = geneLocs_df[(geneLocs_df.begin>(971679-100))&(geneLocs_df.begin<(989858+100))]\n",
    "\n",
    "\n",
    "\n",
    "    testStatiscticsRep1 = pd.merge(small_log_df, small_t_df, how='left', on='locusId')\n",
    "    \n",
    "    pathToData = \"./../RB-TnSeq/2017_11_03_miseq_barseq2/\"\n",
    "    gene_counts_df = pd.read_csv(path.join(pathToData, 'all.poolcount'), sep='\\t')\n",
    "    logratios_df = pd.read_csv(path.join(pathToData, 'fit_logratios.tab'), sep='\\t')\n",
    "    tValues_df = pd.read_csv(path.join(pathToData, 'fit_t.tab'), sep='\\t')\n",
    "    smaller_df = pd.DataFrame()\n",
    "    smaller_df[\"barcode\"] = gene_counts_df[\"barcode\"]\n",
    "    smaller_df[\"locusId\"] = gene_counts_df[\"locusId\"]\n",
    "    smaller_df[\"Initial\"] = gene_counts_df[\"test.S1\"]\n",
    "    smaller_df[\"HighCO2\"] = gene_counts_df[\"test.S2\"]\n",
    "    smaller_df[\"LowCO2\"] = gene_counts_df[\"test.S3\"]\n",
    "\n",
    "    # smaller_df\n",
    "    small_log_df = pd.DataFrame()\n",
    "    small_t_df = pd.DataFrame()\n",
    "    small_log_df[\"locusId\"] = logratios_df[\"locusId\"]\n",
    "    small_log_df[\"sysName\"] = logratios_df[\"sysName\"]\n",
    "    small_log_df[\"desc\"] = logratios_df[\"desc\"]\n",
    "    small_log_df[\"HighCO2Log\"] = logratios_df[\"setAS2 HCO\"]\n",
    "    small_log_df[\"LowCO2Log\"] = logratios_df[\"setAS3 LCO\"]\n",
    "\n",
    "    small_t_df[\"locusId\"] = tValues_df[\"locusId\"]\n",
    "    small_t_df[\"HighCO2T\"] = tValues_df[\"setAS2 HCO\"]\n",
    "    small_t_df[\"LowCO2T\"] = tValues_df[\"setAS3 LCO\"]\n",
    "\n",
    "    testStatiscticsRep2 = pd.merge(small_log_df, small_t_df, how='left', on='locusId')\n",
    "    \n",
    "    means=pd.concat([testStatiscticsRep1,testStatiscticsRep2]).groupby('locusId').mean()\n",
    "#     print(means.head(10))\n",
    "#     print(means.loc['HNEAP_RS04625'])\n",
    "    \n",
    "\n",
    "    labeled_stuff = [\n",
    "        ('HNEAP_RS04625', 'Csos1A',0.11,-4.5),\n",
    "        ('HNEAP_RS04620', 'Csos1B',-0.1,-1.6),\n",
    "        ('HNEAP_RS04630', 'Csos1C',0,-7),\n",
    "        ('HNEAP_RS04635', 'Csos4B',0.1,-5.9),\n",
    "        ('HNEAP_RS04640', 'Csos4A',-1.5,-6.5),\n",
    "        ('HNEAP_RS04650', 'Csos2',0.1,-6.4),\n",
    "        ('HNEAP_RS04660', 'CbbL',-0.9,-5.2),\n",
    "        ('HNEAP_RS04655', 'CbbS',0,-4.1),\n",
    "        ('HNEAP_RS04645', 'CsosCA',0.15,-5),\n",
    "        ('HNEAP_RS04565','Csos1D',-0.05,-1.15),\n",
    "        ('HNEAP_RS01030','DabAB2',0.1,-3.3),\n",
    "#         ('HNEAP_RS01035','DabB2',1,1),\n",
    "        ('HNEAP_RS01040','LysR',0.15,-2.6),\n",
    "        ('HNEAP_RS04615','acRAF',-1.5,-4), #Pterin-4a-carbinolamine dehydratase\n",
    "        ('HNEAP_RS05490','LysR',-0.8,-6),\n",
    "        ('HNEAP_RS07320','Crp/Fnr',0,-2.1),\n",
    "        ('HNEAP_RS04585','DabAB1',-1.5,-2.1),\n",
    "#         ('HNEAP_RS04595','DabB1',1,1),\n",
    "#         ('HNEAP_RS05865','TPR_16',1,1),#Tetratricopeptide TPR_2 repeat protein\n",
    "    #     'HNEAP_RS05380':'Lytic transglycosylase catalytic',\n",
    "    #     'HNEAP_RS08465':'N-acetylmuramyl-L-alanine amidase, negative regulator of AmpC, AmpD'\n",
    "    ]\n",
    "\n",
    "    for gene,label,x,y in labeled_stuff:\n",
    "        plt.text(x,y,label,fontsize=26)\n",
    "    \n",
    "# #     print(means)\n",
    "    \n",
    "    cbbLoc_toPlot = means[means.index.isin(cbbLocus_df.locusId)]\n",
    "\n",
    "#     print(means)\n",
    "    \n",
    "    plt.scatter(means.HighCO2Log, means.LowCO2Log, c=aCol,edgecolors='k', s=100,\n",
    "                label='Outside operon')\n",
    "    plt.scatter(cbbLoc_toPlot.HighCO2Log, cbbLoc_toPlot.LowCO2Log, c=iCol,edgecolors='k', s=100,\n",
    "                label='In extended operon')\n",
    "\n",
    "    \n",
    "    matplotlib.rcParams.update({'font.size': 30})\n",
    "    \n",
    "    \n",
    "    \n",
    "#     plt.scatter(means.HighCO2Log, means.LowCO2Log, c='g',edgecolors='k', s=100)\n",
    "    \n",
    "    plt.xlabel('Fitness in 5% CO$_2$')\n",
    "    plt.ylabel('Fitness in Ambient CO$_2$')\n",
    "    plt.tick_params(axis='both', direction='out', bottom='on', left='on', top='off', right='off',\n",
    "                    length=10, width=4)\n",
    "        # Hide the right and top spines\n",
    "    ax.spines['right'].set_visible(False)\n",
    "    ax.spines['top'].set_visible(False)\n",
    "\n",
    "    # Only show ticks on the left and bottom spines\n",
    "    ax.yaxis.set_ticks_position('left')\n",
    "    ax.xaxis.set_ticks_position('bottom')\n",
    "    \n",
    "#     change axis weight\n",
    "    ax.spines['left'].set_linewidth(3)\n",
    "    ax.spines['bottom'].set_linewidth(3)\n",
    "    locs,labels = plt.xticks()\n",
    "    ticks=['2$^{'+str(int(x))+'}$' for x in locs]\n",
    "    plt.xticks(locs,ticks)#,rotation=45)\n",
    "    locs,labels = plt.yticks()\n",
    "    ticks=['2$^{'+str(int(x))+'}$' for x in locs]\n",
    "    plt.yticks(locs,ticks)\n",
    "        \n",
    "   \n",
    "fig,ax = plt.subplots(figsize=(10,10), frameon=False)\n",
    "makePanelD(ax)\n",
    "plt.savefig('./FigureMakingForBarseqDABPaper/figsForPaper/fig2PanelB.eps',format='eps',bbox_inches='tight')\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true,
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "# Making Panel C\n",
    "%matplotlib inline\n",
    "import matplotlib.pyplot as plt\n",
    "\n",
    "def makePanelC(ax):\n",
    "    import pandas as pd\n",
    "    import numpy as np\n",
    "    import matplotlib as mpl\n",
    "    from os import path\n",
    "    import matplotlib.pyplot as plt\n",
    "    from matplotlib import scale as mscale\n",
    "    from matplotlib import transforms as mtransforms\n",
    "    import matplotlib.cm as cm\n",
    "\n",
    "    import seaborn as sns\n",
    "\n",
    "    cols = ['#56AC52','#337848','#F2E92A','#F58320','#822F31','#CC3C39','#475BA6','#78CDDA','#5375BA','#EA3C81'\n",
    "            ,'#7E441D','#848585','#8E5EA7','#B9D36C','#000000']\n",
    "    colors = sns.color_palette(cols)    \n",
    "#     iCol=colors[0] #non-CCM genes\n",
    "#     eCol=colors[4]  #known CCM genes\n",
    "#     nCol=colors[7] #known non-CCM genes\n",
    "#     aCol=colors[6]  #new genes\n",
    "    \n",
    "    insertColor=colors[0]\n",
    "    essentialColor=colors[4]\n",
    "    nonEssentialColor=colors[5]\n",
    "    aCol=colors[6]\n",
    "\n",
    "    window_size = 300\n",
    "    pool_f = open(\"./../RB-TnSeq/2017_11_03_miseq_barSeq1/EnrichmentWWindow\"+str(window_size), \"r\")\n",
    "    pool_df = pd.read_table(pool_f,sep=',')\n",
    "    pool_f.close()\n",
    "    pool_dropped_df =  pool_df.dropna()\n",
    "    \n",
    "    \n",
    "    inserts_f = open(\"./../RB-TnSeq/2017_11_03_miseq_barSeq1/all.poolcount\", \"r\")\n",
    "    inserts_df = pd.read_table(inserts_f)\n",
    "    inserts_f.close()\n",
    "    \n",
    "    inserts_dropped_df = inserts_df.dropna(subset=['pos'])\n",
    "#     nMax=1.2\n",
    "#     nMin=3.4\n",
    "#     oMax=max(pool_dropped_df['val'])\n",
    "#     oMin=min(pool_dropped_df['val'])\n",
    "#     roll=((pool_dropped_df['val']-oMin)/(oMax-oMin))*(nMax-nMin)+nMin\n",
    "#     zero=((0-oMin)/(oMax-oMin))*(nMax-nMin)+nMin\n",
    "#     pool_dropped_df['val']=roll\n",
    "\n",
    "    rep1 = pd.read_csv('./HnTnSeqAnalysis/geneStatsRep1.csv')\n",
    "    pathToData = \"./../RB-TnSeq/2017_11_03_miseq_barSeq1/\"\n",
    "    logratios_df = pd.read_csv(path.join(pathToData, 'fit_logratios.tab'), sep='\\t')\n",
    "    # smaller_df\n",
    "    small_log_df = pd.DataFrame()\n",
    "    small_log_df[\"locusId\"] = logratios_df[\"locusId\"]\n",
    "    small_log_df[\"sysName\"] = logratios_df[\"sysName\"]\n",
    "    small_log_df[\"desc\"] = logratios_df[\"desc\"]\n",
    "    small_log_df[\"HighCO2Log\"] = logratios_df[\"setAS2 HCO\"]\n",
    "    small_log_df[\"LowCO2Log\"] = logratios_df[\"setAS3 LCO\"]\n",
    "    small_log_df['enrichment']=small_log_df.LowCO2Log-small_log_df.HighCO2Log\n",
    "    \n",
    "    height=1\n",
    "#     region=[(983594-13000,986794+8000),(216936-3000,219419+6000)]\n",
    "    \n",
    "    region=[(983594-14000,986794+11000),(216936-6000,219419+9000)]\n",
    "#     print(rep1)\n",
    "    region.sort()\n",
    "    print(region)\n",
    "    positions=[]\n",
    "    for reg in region:\n",
    "    #     print(reg)\n",
    "        positions.extend(list(range(reg[0],reg[1])))\n",
    "        positions.extend([np.nan]*50)\n",
    "    positions=np.array(positions)\n",
    "    positions_dropped = positions[~np.isnan(positions)]\n",
    "\n",
    "    \n",
    "    def CustomScaleFactory(l, u):\n",
    "        class CustomScale(mscale.ScaleBase):\n",
    "            name = 'custom'\n",
    "\n",
    "            def __init__(self, axis, **kwargs):\n",
    "                mscale.ScaleBase.__init__(self)\n",
    "                self.thresh = None #thresh\n",
    "\n",
    "            def get_transform(self):\n",
    "                return self.CustomTransform(self.thresh)\n",
    "\n",
    "            def set_default_locators_and_formatters(self, axis):\n",
    "                pass\n",
    "\n",
    "            class CustomTransform(mtransforms.Transform):\n",
    "                input_dims = 1\n",
    "                output_dims = 1\n",
    "                is_separable = True\n",
    "                lower = l\n",
    "                upper = u\n",
    "                def __init__(self, thresh):\n",
    "                    mtransforms.Transform.__init__(self)\n",
    "                    self.thresh = thresh\n",
    "\n",
    "                def transform_non_affine(self, a):\n",
    "                    aa=np.array(a)\n",
    "                    aaa = np.array(a)\n",
    "                    aaa[aa>self.lower] = aa[aa>self.lower]-(self.upper-self.lower)\n",
    "                    aaa[(aa>self.lower)&(aa<self.upper)] = self.lower\n",
    "                    return aaa\n",
    "\n",
    "                def inverted(self):\n",
    "                    return CustomScale.InvertedCustomTransform(self.thresh)\n",
    "\n",
    "            class InvertedCustomTransform(mtransforms.Transform):\n",
    "                input_dims = 1\n",
    "                output_dims = 1\n",
    "                is_separable = True\n",
    "                lower = l\n",
    "                upper = u\n",
    "\n",
    "                def __init__(self, thresh):\n",
    "                    mtransforms.Transform.__init__(self)\n",
    "                    self.thresh = thresh\n",
    "\n",
    "                def transform(self, a):\n",
    "                    aa=np.array(a)\n",
    "                    aaa = np.array(a)\n",
    "                    aaa[aa>self.lower] = aaa[aa>self.lower]+(self.upper-self.lower)\n",
    "                    return aaa\n",
    "\n",
    "                def inverted(self):\n",
    "                    return CustomScale.CustomTransform(self.thresh)\n",
    "\n",
    "        return CustomScale\n",
    "\n",
    "    mscale.register_scale(CustomScaleFactory(region[0][1]+2000, region[1][0]-2000))\n",
    "    plt.gca().set_xscale('custom')\n",
    "    \n",
    "    \n",
    "    y=0.45\n",
    "    rolMed=[0]*(len(positions_dropped))\n",
    "    geneRegion = rep1[((rep1.begin.isin(positions_dropped))&(rep1.end.isin(positions_dropped)))]\n",
    "    print(geneRegion.locusId)\n",
    "    print(geneRegion.desc)\n",
    "#     print(pool_dropped_df)\n",
    "#     print(pool_dropped_df.iloc[:,1])\n",
    "#     print(pool_dropped_df['position'])\n",
    "    insertRegion = pool_dropped_df[pool_dropped_df.position.isin(positions_dropped)]\n",
    "    poses = inserts_dropped_df[inserts_dropped_df.pos.isin(positions_dropped)].pos\n",
    "    width=0.05\n",
    "    headwidth=0.085\n",
    "    height = 0.07\n",
    "    head_scale=1\n",
    "    \n",
    "#     smaller_log_df=small_log_df[small_log_df.locusId.isin(geneRegion.locusId)]\n",
    "#     norm = mpl.colors.Normalize(vmin=min(smaller_log_df['enrichment']), vmax=max(smaller_log_df['enrichment']))\n",
    "\n",
    "    norm = mpl.colors.Normalize(vmin=min(small_log_df['enrichment']), vmax=max(small_log_df['enrichment']))\n",
    "#     cmap = sns.diverging_palette(240,10,as_cmap=True)\n",
    "    cmap = sns.cubehelix_palette(rot=-.4,as_cmap=True,reverse=True)\n",
    "    normMap = cm.ScalarMappable(norm=norm, cmap=cmap)\n",
    "    for i,gene in enumerate(geneRegion.index):\n",
    "\n",
    "        length = geneRegion.loc[gene,'length']\n",
    "        \n",
    "        hl=500\n",
    "        if(length>hl):\n",
    "            head_length = hl\n",
    "        else:\n",
    "            head_length=length*head_scale\n",
    "#         print(geneRegion.loc[gene,'locusId'])\n",
    "#         print(small_log_df[small_log_df.locusId == geneRegion.loc[gene,'locusId']])\n",
    "#         enrichment=small_log_df[small_log_df.locusId.str.contains(geneRegion.loc[gene,'locusId'])]['enrichment']\n",
    "#         print(enrichment)\n",
    "#         c = normMap.to_rgba([enrichment])\n",
    "#         print(c)\n",
    "#         c=list(c)\n",
    "#         print(c)\n",
    "        if(sum(small_log_df.locusId.str.contains(geneRegion.loc[gene,'locusId']))):\n",
    "            enrichment=small_log_df[small_log_df.locusId.str.contains(geneRegion.loc[gene,'locusId'])]['enrichment']\n",
    "#             print(enrichment)\n",
    "            c = normMap.to_rgba(enrichment)[0]\n",
    "#             print(c)\n",
    "        else:\n",
    "#             print(nonEssentialColor)\n",
    "            c=nonEssentialColor\n",
    "\n",
    "        if(geneRegion.loc[gene,'strand']=='+'):\n",
    "            start=(geneRegion.loc[gene,'begin'])\n",
    "            genePic = mpl.patches.FancyArrow(start,y,length,\n",
    "                                             0,width=width,length_includes_head=True,head_width=headwidth,\n",
    "                                             head_length=head_length, fill=True,facecolor=c)\n",
    "            ax.add_artist(genePic)\n",
    "        else:\n",
    "            start=geneRegion.loc[gene,'end']\n",
    "            genePic = mpl.patches.FancyArrow(start,y,-length,\n",
    "                                             0,width=width,length_includes_head=True,head_width=headwidth,\n",
    "                                             head_length=head_length, fill=True,facecolor=c)\n",
    "            ax.add_artist(genePic)\n",
    "    normMap.set_array([])\n",
    "    plt.colorbar(mappable=normMap,orientation='horizontal')\n",
    "#     plt.ylim(0.2,0.8)\n",
    "    plt.ylim(0.36,0.6)\n",
    "    \n",
    "#     nMax=1.2\n",
    "#     nMin=3.4\n",
    "#     oMin=min(insertRegion['EnrichmentWWindow300'])\n",
    "#     oMax=max(insertRegion['EnrichmentWWindow300'])\n",
    "\n",
    "#     roll=((insertRegion['EnrichmentWWindow300']-oMin)/(oMax-oMin))*(nMax-nMin)+nMin\n",
    "#     zero=np.array([((0-oMin)/(oMax-oMin))*(nMax-nMin)+nMin]*2)\n",
    "#     plt.plot(insertRegion['position'],(height*roll)+(y),'.',color=aCol)\n",
    "#     plt.plot(region[0],(height*zero)+(y),color=essentialColor)\n",
    "#     plt.plot(region[1],(height*zero)+(y),color=essentialColor)\n",
    "#         plt.vlines(bases,[y+height*2.3]*len(bases),\n",
    "#                    ([y+height*2.3]*len(bases))+(height*2*rolMed), colors='k')\n",
    "\n",
    "    plt.vlines(poses,[y+height*0.8]*len(poses),\n",
    "               ([y+height*1.4]*len(poses)), colors=insertColor,linewidth=0.75)\n",
    "    plt.xticks([])\n",
    "    \n",
    "    plt.scatter([region[0][1]+1500,region[0][1]+2000,region[1][0]-1500],[y+headwidth/2]*3,color=colors[14])\n",
    "#     plt.plot([region[0][1]+1500,region[0][1]+2000],[y+height*0.8,y+height*1.4],color='k',lw=2)\n",
    "#     plt.plot([region[1][0]-2000,region[1][0]-1500],[y+height*0.8,y+height*1.4],color='k',lw=2)\n",
    "    \n",
    "#     plt.plot([region[0][1]+1500,region[0][1]+2000],[y-width/2,y+width/2],color='k',lw=2)\n",
    "#     plt.plot([region[1][0]-2000,region[1][0]-1500],[y-width/2,y+width/2],color='k',lw=2)\n",
    "    \n",
    "    plt.xlim(min(region[0]),max(region[1]))\n",
    "    ax.xaxis.set_visible(False)\n",
    "    ax.yaxis.set_visible(False)\n",
    "    ax.spines['right'].set_visible(False)\n",
    "    ax.spines['top'].set_visible(False)\n",
    "    ax.spines['left'].set_visible(False)\n",
    "    ax.spines['bottom'].set_visible(False)\n",
    "    \n",
    "fig,ax = plt.subplots(figsize=(20,3), frameon=False)\n",
    "makePanelC(ax)\n",
    "plt.savefig('./FigureMakingForBarseqDABPaper/figsForPaper/fig2PanelC.eps',\n",
    "           format='eps',bbox_inches='tight')\n",
    "plt.show()"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.4.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
